### PR TITLE
new feature: auto detect snippet language

### PR DESCRIPTION
**Pull request by elfman**
## Description

for #2351 
add new feature: auto detect snippet language
try to detect after pasting and mode has not been set

detect

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- ⚪️ : I have written test for my code and it has been tested
- :radio_button: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

**Comment by elfman** [comment]
@ZeroX-DG my pr can not pass the check, but what it figures out has nothing to do with what I want to change.
Should I fix it in this pr?

**Comment by ZeroX-DG** [comment]
@Rokt33r Should he fix it in this PR or just leave it and you can open another PR to fix it?

File: browser/components/CodeEditor.js [hunk 1]
```diff
@@ -274,7 +354,11 @@ export default class CodeEditor extends React.Component {
       extraKeys: this.defaultKeyMap
     })
 
-    this.setMode(this.props.mode)
+    if (!this.props.mode && this.props.value) {
```
Comment (ZeroX-DG) [inline browser/components/CodeEditor.js]: Should you check for `config.editor.snippetDefaultLanguage === 'Auto Detect'` here too?

Comment (elfman) [reply browser/components/CodeEditor.js]: Updated

**Pull request merged**